src/Tizen.Multimedia.Remoting/MediaController/MediaControlCommand.cs

```diff
@@ -334,16 +343,181 @@ internal override string Request()
+                        condition.Keyword, condition.Bundle.SafeBundleHandle).
+                        ThrowIfError("Failed to set search condition.");
+                }
+                else
+                {
+                    NativeClient.SetSearchCondition(_searchHandle, condition.ContentType,
+                        condition.Category == 0 ? MediaControlNativeSearchCategory.NoCategory : condition.Category.ToNative(),
+                        condition.Keyword, IntPtr.Zero).
+                        ThrowIfError("Failed to set search condition.");
+                }
+            }
+            catch
+            {
+                if (_searchHandle != IntPtr.Zero)
+                {
+                    NativeClient.DestroySearchHandle(_searchHandle).ThrowIfError("Failed to destroy search handle");
+                }
+            }
+        }
+
+        internal SearchCommand(List<MediaControlSearchCondition> conditions, IntPtr searchHandle)
+        {
+            _searchHandle = searchHandle;
+
+            try
+            {
+                foreach (var condition in conditions)
+                {
+                    if (condition.Bundle != null)
+                    {
+                        NativeClient.SetSearchConditionBundle(_searchHandle, condition.ContentType,
+                            condition.Category == 0 ? MediaControlNativeSearchCategory.NoCategory : condition.Category.ToNative(),
+                            condition.Keyword, condition.Bundle.SafeBundleHandle).
+                            ThrowIfError("Failed to set search condition.");
+                    }
+                    else
+                    {
+                        NativeClient.SetSearchCondition(_searchHandle, condition.ContentType,
+                            condition.Category == 0 ? MediaControlNativeSearchCategory.NoCategory : condition.Category.ToNative(),
+                            condition.Keyword, IntPtr.Zero).
+                            ThrowIfError("Failed to set search condition.");
+                    }
+                }
+            }
+            catch
+            {
+                if (_searchHandle != IntPtr.Zero)
+                {
+                    NativeClient.DestroySearchHandle(_searchHandle).ThrowIfError("Failed to destroy search handle");
+                }
+            }
+        }
+
+        internal override string Request()
+        {
+            NativeClient.SendSearchCommand(_clientHandle, ServerId, _searchHandle, out string requestId).
+                ThrowIfError("Failed to send search command.");
+
+            if (_searchHandle != IntPtr.Zero)
+            {
+                NativeClient.DestroySearchHandle(_searchHandle).ThrowIfError("Failed to destroy search handle");
+            }
+
+            return requestId;
+        }
+
+        internal override void Response(IntPtr handle, int result, Bundle bundle)
```

Comment:
It is not good design, It is bad to OCP.
SubClass of Command need not to know detail of Response implementation

I recommand add a protected method `OnResponseCompleted` on Command class
``` c#
class abstract class Command {
...
 protected virtual OnResponseCompleted() {}
 internal void Response(IntPtr handle, int result, Bundle bundle)
 {
     try {
         if (bundle != null)
         {
             NativeServer.SendCommandReplyBundle(handle, _clientId, _requestId, result, bundle.SafeBundleHandle)
                 .ThrowIfError("Failed to response command.");
          }
          else
          {
              NativeServer.SendCommandReply(handle, _clientId, _requestId, result, IntPtr.Zero)
                  .ThrowIfError("Failed to response command.");
           }
      }
      finally
      {
           OnResponseCompleted();
      }
 }
...
}
```
And override `OnResponseCompleted` in SearchCommand class
``` c#
class SearchCommand {
  ...
 protected override OnResponseCompleted() {
   if (_searchHandle != IntPtr.Zero)
   {
        NativeClient.DestroySearchHandle(_searchHandle).ThrowIfError("Failed to destroy search handle");
    }
 }
  ...
}
```

Reply:
Ok. I'll change it.